### PR TITLE
VACMS-14699: Adding a Entity Field Fetch fields for VBA Centralized content

### PR DESCRIPTION
**Pull request by omahane**
## Description

Relates to #14699

## Testing done
Manually

### node:view
Screenshot 2023-10-04 at 2 13 27 PM

### node:edit
Screenshot 2023-10-04 at 2 15 26 PM

### Reviewing node:view of a VBA facility
- [ ] As an admin, visit to [a VBA facility](https://pr15528-0squmrsgqim6fn8sjpgkwzrlslxcq25u.ci.cms.va.gov/vba-facilities/locations/fort-harrison-va-regional-benefit-satellite-office-at-david-j-thatcher-va-clinic) in node:view
- [ ] Confirm that the following fields are being fetched:
    - [ ] Benefit office overview
    - [ ] Benefits hotline
    - [ ] Benefit office hours may vary
    - [ ] National spotlight 1
    - [ ] National spotlight 2
    - [ ] National spotlight 3
    - [ ] Can't find benefits
    - [ ] Get updates from VBA
- [ ] Confirm that each fetched field is in a grey box
- [ ] Confirm that each fetched field has a tooltip that pops up when hovering over the question mark

### Reviewing node:edit of a VBA facility
- [ ] As an admin, edit [a VBA facility]https://pr15528-0squmrsgqim6fn8sjpgkwzrlslxcq25u.ci.cms.va.gov/node/61682/edit)
- [ ] Confirm that the following fields are being fetched:
    - [ ] Benefit office overview
    - [ ] Benefits hotline
    - [ ] Benefit office hours may vary
    - [ ] National spotlight 1
    - [ ] National spotlight 2
    - [ ] National spotlight 3
    - [ ] Can't find benefits
    - [ ] Get updates from VBA
- [ ] Confirm that each fetched field is in a grey box
- [ ] Confirm that each fetched field has a tooltip that pops up when hovering over the question mark
- [ ] Make a change to the facility
- [ ] Save
- [ ] Confirm that the fetched fields and change you made are showing on node:view

### Editing VBA Centralized Content
- [ ] As an editor, edit the [National VA benefit office content](https://pr15528-0squmrsgqim6fn8sjpgkwzrlslxcq25u.ci.cms.va.gov/node/61744/edit)
- [ ] Make a change to the content in the "Rich text - char limit 1000" paragraph under the "Benefit office overview" Centralized content descriptor
- [ ] Save 
- [ ] Visit to [a VBA facility](https://pr15528-0squmrsgqim6fn8sjpgkwzrlslxcq25u.ci.cms.va.gov/vba-facilities/locations/fort-harrison-va-regional-benefit-satellite-office-at-david-j-thatcher-va-clinic) in node:view
- [ ] Confirm that your change is reflected in the "Benefit office overview" field

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### CODE REVIEW

File: config/sync/field.field.node.vba_facility.field_cc_vba_facility_overview.yml
Comment (davidmpickett) [inline config/sync/field.field.node.vba_facility.field_cc_vba_facility_overview.yml:1]: I would expect this to be named field_cc_benefit_office_overview based on the label I suggested

File: config/sync/core.entity_view_display.node.vba_facility.default.yml
Comment (davidmpickett) [inline config/sync/core.entity_view_display.node.vba_facility.default.yml:1]: >     - field.field.node.vba_facility.field_benefit_office_hours_may_v
>     - field.field.node.vba_facility.field_can_t_find_benefits
>     - field.field.node.vba_facility.field_get_updates_from_vba
>     - field.field.node.vba_facility.field_national_spotlight_1
>     - field.field.node.vba_facility.field_national_spotlight_2
>     - field.field.node.vba_facility.field_national_spotlight_3

  Why don't these have _cc_ prepended?

File: config/sync/field.field.node.vba_facility.field_can_t_find_benefits.yml
Comment (davidmpickett) [inline config/sync/field.field.node.vba_facility.field_can_t_find_benefits.yml:1]: > field_can_t_find_benefits

Not sure if there's an official style guide for machine names, but I would not replace an apostrophe with an underscore
field_cant_find_benefits

File: config/sync/field.field.node.vba_facility.field_benefit_office_hours_may_v.yml
Comment (davidmpickett) [inline config/sync/field.field.node.vba_facility.field_benefit_office_hours_may_v.yml:1]: > field_benefit_office_hours_may_v

Consider: field_benefit_office_hours_vary

File: config/sync/field.field.node.vba_facility.field_can_t_find_benefits.yml
Comment (omahane) [reply config/sync/field.field.node.vba_facility.field_can_t_find_benefits.yml:1]: Resolved.

File: config/sync/core.entity_view_display.node.vba_facility.default.yml
Comment (omahane) [reply config/sync/core.entity_view_display.node.vba_facility.default.yml:1]: Resolved.

File: config/sync/field.field.node.vba_facility.field_cc_vba_facility_overview.yml
Comment (omahane) [reply config/sync/field.field.node.vba_facility.field_cc_vba_facility_overview.yml:1]: Resolved

File: config/sync/core.entity_form_display.node.vba_facility.default.yml [hunk 1]
```diff
@@ -110,8 +119,10 @@ third_party_settings:
         - field_address
         - field_geolocation
         - field_phone_number
+        - group_benefits_hotline
         - field_office_hours
         - field_timezone
+        - group_benefit_office_hours_may_v
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.vba_facility.default.yml:125]: I think I am with Dave's suggestion on this 
```suggestion
        - group_benefit_office_hours_vary
```

File: config/sync/core.entity_form_display.node.vba_facility.default.yml [hunk 2]
```diff
@@ -163,6 +174,127 @@ third_party_settings:
...
+        classes: centralized
+        element: div
+        show_label: '0'
+        label_element: h3
+        required_fields: 1
+        attributes: ''
+        description: ''
+        tooltip_description: "Why can’t I edit this?\r\nInternational editors keep this content standardized to provide consistent messaging for facilities nationwide."
+        open: false
+    group_benefit_office_hours_may_v:
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.vba_facility.default.yml:216]:
```suggestion
    group_benefit_office_hours_vary:
```

File: config/sync/core.entity_view_display.node.vba_facility.default.yml [hunk 3]
```diff
@@ -68,7 +77,9 @@ third_party_settings:
         - field_facility_locator_api_id
         - field_address
         - field_phone_number
+        - group_benefits_hotline
         - field_office_hours
+        - group_benefit_office_hours_may_v
```
Comment (swirtSJW) [inline config/sync/core.entity_view_display.node.vba_facility.default.yml:82]:
```suggestion
        - group_benefit_office_hours_vary
```

File: config/sync/core.entity_view_display.node.vba_facility.default.yml [hunk 4]
```diff
@@ -110,14 +121,134 @@ third_party_settings:
...
+        classes: centralized
+        element: div
+        show_label: '0'
+        label_element: h3
+        attributes: ''
+        description: ''
+        tooltip_description: "Why can’t I edit this?\r\nNational editors keep this content standardized to provide consistent messaging for facilities nationwide."
+        open: false
+        required_fields: false
+    group_benefit_office_hours_may_v:
```
Comment (swirtSJW) [inline config/sync/core.entity_view_display.node.vba_facility.default.yml:170]:
```suggestion
    group_benefit_office_hours_vary:
```

**Review by swirtSJW — COMMENTED** [review]
One set of suggested changes to make the machine name more friendly and meaningful for front end.

File: config/sync/core.entity_form_display.node.vba_facility.default.yml (hunk 1 above)
Comment (omahane) [reply config/sync/core.entity_form_display.node.vba_facility.default.yml]: I must have missed this. Done.

**Comment by swirtSJW** [comment]
This placement seems odd putting the centralized content inside the facility data section
image

Because it is not facility data and makes it look like it is coming from elsewhere.

**Comment by swirtSJW** [comment]
Is it intentional that we are only showing the last updated date on once of the CC fields but not the other?
image
image

The date it going to be the same for all of them.. but it shows the source of each segment of the data.

**Comment by swirtSJW** [comment]
Strikes me odd that these are rendering as separate link title and urls.   It is going to be based on the default view mode for that paragraph.... which might not have had any attention when it was built.   
image

I think it is unrelated to this ticket.  Just noting it is odd.

**Comment by omahane** [comment]
> This placement seems odd putting the centralized content inside the facility data section image
> 
> Because it is not facility data and makes it look like it is coming from elsewhere.

I was trying to [follow the layout from the FE](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/13799#issuecomment-1703537306), which sandwiches this number right after the main number and before the hours  for editor clarity, but if this is confusing, I could put it elsewhere. After?

Screenshot 2023-10-05 at 2 43 41 PM

**Comment by omahane** [comment]
> Is it intentional that we are only showing the last updated date on once of the CC fields but not the other?
> The date it going to be the same for all of them.. but it shows the source of each segment of the data.

This was unintentional. @davidmpickett I think I should include the updated date. What do you think?

**Comment by davidmpickett** [comment]
> > Is it intentional that we are only showing the last updated date on once of the CC fields but not the other?
> > The date it going to be the same for all of them.. but it shows the source of each segment of the data.
> 
> This was unintentional. @davidmpickett I think I should include the updated date. What do you think?

I agree so heartily that I fleshed out a whole ticket about how I think this should be the default pattern: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/4055

**Comment by swirtSJW** [comment]
> I agree so heartily that I fleshed out a whole ticket about how I think this should be the default pattern: #4055

It is all about ticking all the boxed :grinning: 
image

**Review by swirtSJW — APPROVED** [review]
Looks awesome.   Nice work and thanks for sticking it out through all the changes.

**Comment by swirtSJW** [comment]
image

**Pull request merged**